### PR TITLE
Fix (count_with_limit): Make grouping configurable

### PR DESCRIPTION
**Pull request by OjusWiZard**
## Checklist

- [x] For counting with limit, only do the group by sub-query if `group_by_event_ids` is True.

### CODE REVIEW

File: rotkehlchen/db/history_events.py [hunk 1]
```diff
@@ -628,12 +628,15 @@ def get_history_events_count(
             )
 
             free_query_group_by = maybe_filter_ignore_asset(query_filter)
+            group_by_query = ''
+            if group_by_event_ids:
+                group_by_query = 'GROUP BY event_identifier ORDER BY timestamp DESC, sequence_index ASC LIMIT ?'  # noqa: E501
```
Comment (yabirgb) [inline rotkehlchen/db/history_events.py:633]: how does this change affect the frontend? Does it break something in their assumptions? It should be documented in the PR description if it is changing the default behaviour

File: rotkehlchen/tests/api/test_history_base_entry.py [hunk 2]
```diff
@@ -391,7 +391,7 @@ def test_get_events(rotkehlchen_api_server: 'APIServer'):
         ),
     )
     result = assert_proper_response_with_result(response)
-    assert result['entries_found'] == 6
+    assert result['entries_found'] == 9
```
[the remark this answers is not in the document]
Comment (OjusWiZard) [reply rotkehlchen/tests/api/test_history_base_entry.py:394]: it's there just below this one, on the line 418.

File: rotkehlchen/db/history_events.py (hunk 1 above)
Comment (OjusWiZard) [reply rotkehlchen/db/history_events.py:633]: I tested it by running the app, and all the paginated events were showing as expected. Also, I've asked the frontend devs, if they ever use `group_by_event_ids: false` on this API.

Comment (OjusWiZard) [reply rotkehlchen/db/history_events.py:633]: @tewshi confirmed that it's not used.

**Review by LefterisJP — APPROVED** [review]
LGTM

**Pull request merged**